gpt_researcher/master/agent.py

```diff
@@ -68,17 +84,13 @@ async def run_sub_query(self, sub_query):
         """
         # Get Urls
         retriever = self.retriever(sub_query)
-        urls = retriever.search()
-        urls_to_scrape = []
-        for url in urls:
-            if url not in self.visited_urls:
-                await self.stream_output("logs", f"✅ Adding source url to research: {url}\n")
-                self.visited_urls.add(url)
-                urls_to_scrape.append(url)
+        search_results = retriever.search()
+        new_search_urls = await self.get_new_urls([url.get("href") for url in search_results])
```

Comment:
Seems like this line works only for tavily, duckduckgo's response is just a list of url's, maybe we should add a new_search_urls method to the retriever class.